pkg/scheduler/partition.go

```diff
@@ -49,6 +49,7 @@ type PartitionContext struct {
 	// Private fields need protection
 	root                   *objects.Queue                  // start of the queue hierarchy
 	applications           map[string]*objects.Application // applications assigned to this partition
+	completedApplications  map[string]*objects.Application // completed applications from this partition
```

Comment:
let's remove this map, such overhead seems not to be necessary
because the app cleanup happens in the background, we can iterate the application's map and delete apps that has expired, it won't cause any perf issue